pkg/lib/indexer/indexer_test.go

```diff
@@ -84,3 +85,71 @@ func TestGeneratePackageYaml(t *testing.T) {
 
 	_ = os.RemoveAll("./package.yaml")
 }
+
+func TestEnsureCSVFields(t *testing.T) {
+	tests := []struct {
+		name   string
+		values bundleDirPrefix
+		csv    string
+		want   string
+	}{
+		{
+			name: "don't add replaces when it is already an existing skips entry",
+			values: bundleDirPrefix{
+				replaces: "v0.0.1",
+			},
+			csv:  `{"kind":"ClusterServiceVersion","metadata":{"name":"testoperator.v1.0.0"},"spec":{"installModes":[{"supported":true,"type":"OwnNamespace"}],"skips":["v0.0.0","v0.0.1","v0.0.1-1"],"version":"v1.0.0"}}`,
+			want: `{"kind":"ClusterServiceVersion","metadata":{"name":"testoperator.v1.0.0"},"spec":{"installModes":[{"supported":true,"type":"OwnNamespace"}],"skips":["v0.0.0","v0.0.1","v0.0.1-1"],"version":"v1.0.0"}}`,
+		},
+		{
+			name: "add new replaces entry",
+			values: bundleDirPrefix{
+				replaces: "v0.0.2",
+			},
+			csv:  `{"kind":"ClusterServiceVersion","metadata":{"name":"testoperator.v1.0.0"},"spec":{"installModes":[{"supported":true,"type":"OwnNamespace"}],"skips":["v0.0.0","v0.0.1","v0.0.1-1"],"version":"v1.0.0"}}`,
+			want: `{"kind":"ClusterServiceVersion","metadata":{"name":"testoperator.v1.0.0"},"spec":{"installModes":[{"supported":true,"type":"OwnNamespace"}],"replaces":"v0.0.2","skips":["v0.0.0","v0.0.1","v0.0.1-1"],"version":"v1.0.0"}}`,
+		},
+		{
+			name: "update replaces entry",
+			values: bundleDirPrefix{
+				replaces: "v0.0.3",
+			},
+			csv:  `{"kind":"ClusterServiceVersion","metadata":{"name":"testoperator.v1.0.0"},"spec":{"installModes":[{"supported":true,"type":"OwnNamespace"}],"replaces":"v0.0.2","skips":["v0.0.0","v0.0.1","v0.0.1-1"],"version":"v1.0.0"}}`,
+			want: `{"kind":"ClusterServiceVersion","metadata":{"name":"testoperator.v1.0.0"},"spec":{"installModes":[{"supported":true,"type":"OwnNamespace"}],"replaces":"v0.0.3","skips":["v0.0.0","v0.0.1","v0.0.1-1","v0.0.2"],"version":"v1.0.0"}}`,
+		},
+		{
+			name: "update skips list",
+			values: bundleDirPrefix{
+				skips: map[string]struct{}{
+					"v0.0.1": {},
+					"v0.0.2": {},
+					"v0.0.3": {},
+				},
+			},
+			csv:  `{"kind":"ClusterServiceVersion","metadata":{"name":"testoperator.v0.0.1"},"spec":{"installModes":[{"supported":true,"type":"OwnNamespace"}],"replaces":"v0.0.3","skips":["v0.0.0","v0.0.2"],"version":"1.0.0"}}`,
+			want: `{"kind":"ClusterServiceVersion","metadata":{"name":"testoperator.v0.0.1"},"spec":{"installModes":[{"supported":true,"type":"OwnNamespace"}],"replaces":"v0.0.3","skips":["v0.0.0","v0.0.1","v0.0.2"],"version":"1.0.0"}}`,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f, err := os.OpenFile("testdata/testoperator/testoperator.csv.yaml", os.O_WRONLY|os.O_TRUNC, 0)
```

Comment:
If the function under test writes to files in the target directory, the test should set up a temporary file/directory to work under. As a reader, I'm going to expect anything in testdata to be a static test fixture. Alternatively, you could refactor the function under test so that it doesn't directly deal with files.